gr-blocks/lib/nlog10_ff_impl.cc

```diff
@@ -54,8 +58,17 @@ namespace gr {
       float n = d_n;
       float k = d_k;
 
-      for (int i = 0; i < noi; i++)
-        out[i] = n * log10f(std::max(in[i], (float) 1e-18)) + k;
+      volk_32f_log2_32f(out, in, noi);
+      volk_32f_s32f_multiply_32f(out, out, n/log2f(10.0f), noi);
```

Comment:
The value "n/log2f(10.0f)" can be precomputed and kept in a class variable d_n_log2_10 which would be updated when d_n is updated.